module/platform/NUSense/HardwareIO/src/HardwareIO.cpp

```diff
@@ -85,11 +86,74 @@ namespace module::platform::NUSense {
 
+            // TODO: Apply the servo offsets within NUSense instead
             // Apply servo offsets
             for (size_t i = 0; i < config["servos"].config.size(); ++i) {
                 nugus.servo_offset[i]    = config["servos"][i]["offset"].as<Expression>();
                 nugus.servo_direction[i] = config["servos"][i]["direction"].as<Expression>();
+
+                cfg.servo_configurations[i].direction = config["servos"][i]["direction"].as<int32_t>();
+                cfg.servo_configurations[i].offset = config["servos"][i]["offset"].as<double>();
             }
+
+            handshake_watchdog =
+                on<Watchdog<HandshakeWatchdog, 1, std::chrono::seconds>, Sync<HandshakeWatchdog>>().then(
+                    "Sending handshake message to NUSense",
+                    [this] {
+                        // Parse handshake packet
+                        auto handshake = NUSenseHandshake();
+
+                        handshake.msg = "";
+
+                        for (size_t i = 0; i < cfg.servo_configurations.size(); ++i) {
+                            handshake.servo_configs.emplace_back(cfg.servo_configurations[i].direction,
+                                                                 cfg.servo_configurations[i].offset);
+                        }
+
+                        // Send to stream reactor
+                        send_packet(handshake);
+
+                        log<INFO>("Handshake message sent to NUSense, waiting for response...");
+                    });
+
+            // Sync is used because uart write is a shared resource
+            NUC_packet_catcher =
+                on<Trigger<ServoTargets>, Sync<HardwareIO>>().then([this](const ServoTargets& commands) {
+                    // Copy the data into a new message so we can use a duration instead of a timepoint
+                    // and take the offsets and switch the direction.
+                    auto servo_targets = SubcontrollerServoTargets();
+
+                    // Change the timestamp in servo targets to the difference between the timestamp and now
+                    // Take away the offset and switch the direction if needed
+                    for (auto& target : commands.targets) {
+                        servo_targets.targets.emplace_back(
+                            target.time - NUClear::clock::now(),
+                            target.id,
+                            (target.position - nugus.servo_offset[target.id]) * nugus.servo_direction[target.id],
+                            target.gain,
+                            target.torque);
+                    }
+
+                    send_packet(servo_targets);
+                });
+
+            NUC_packet_catcher.disable();
+        });
+
+        on<Startup>().then([this] {
+            // Start the handshake watchdog
+            emit<Scope::WATCHDOG>(ServiceWatchdog<HandshakeWatchdog>());
+        });
```

Comment:
Watchdogs are active as soon as they're defined, you don't emit a service message to start them or anything. They just run once the time has elapsed and no service has happened. This would prolong the watchdog from running, which doesn't seem to be what we want to do.